packages/components/src/internal/components/chart/ChartBuilderModal.tsx

```diff
@@ -810,3 +872,135 @@ export const ChartBuilderModal: FC<ChartBuilderModalProps> = memo(({ actions, mo
         </Modal>
     );
 });
+
+interface TrendlineOptionProps {
+    fieldValues: Record<string, any>;
+    onFieldChange: (key: string, value: any) => void;
+    schemaQuery: SchemaQuery;
+}
+
+const TrendlineOption: FC<TrendlineOptionProps> = memo(props => {
+    const TRENDLINE_OPTIONS: TrendlineType[] = useMemo(
+        () => TRENDLINE_VALUES.map(value => LABKEY_VIS.GenericChartHelper.TRENDLINE_OPTIONS[value]),
+        []
+    );
+    const { fieldValues, onFieldChange, schemaQuery } = props;
+
+    // hide the trendline option if no x-axis value selected and for date field selection on x-axis
+    const hidden = useMemo(() => {
+        const jsonType = fieldValues.x?.data?.jsonType || fieldValues.x?.data?.type;
+        return !fieldValues.x?.value || jsonType === 'date' || jsonType === 'time';
+    }, [fieldValues.x]);
+
+    const [loadingTrendlineOptions, setLoadingTrendlineOptions] = useState<boolean>(true);
+    const [asymptoteMin, setAsymptoteMin] = useState<string>('');
+    const [asymptoteMax, setAsymptoteMax] = useState<string>('');
+    useEffect(() => {
+        if (loadingTrendlineOptions && (!!fieldValues.trendlineAsymptoteMin || !!fieldValues.trendlineAsymptoteMax)) {
+            setAsymptoteMin(fieldValues.trendlineAsymptoteMin);
+            setAsymptoteMax(fieldValues.trendlineAsymptoteMax);
+            setLoadingTrendlineOptions(false);
+        }
+    }, [fieldValues, loadingTrendlineOptions]);
+
+    const onTrendlineAsymptoteMin = useCallback((event: any) => {
+        setAsymptoteMin(event.target.value);
+    }, []);
+
+    const onTrendlineAsymptoteMax = useCallback((event: any) => {
+        setAsymptoteMax(event.target.value);
+    }, []);
+
+    const applyTrendlineAsymptote = useCallback(() => {
+        onFieldChange('trendlineAsymptoteMin', asymptoteMin);
+        onFieldChange('trendlineAsymptoteMax', asymptoteMax);
+    }, [onFieldChange, asymptoteMin, asymptoteMax]);
+
+    const onTrendlineFieldChange = useCallback(
+        (key: string, _: any, selectedOption: SelectInputOption) => {
```

Comment:
I tried this but then `onTrendlineFieldChange('trendlineType', undefined, undefined);` fails below because:
"Argument of type 'undefined' is not assignable to parameter of type 'never'."